ext/em.cpp

```diff
@@ -592,12 +592,12 @@ void EventMachine_t::_RunEpollOnce()
 	#ifdef HAVE_RB_WAIT_FOR_SINGLE_FD
 	if ((ret = rb_wait_for_single_fd(epfd, RB_WAITFD_IN|RB_WAITFD_PRI, &tv)) < 1) {
 	#else
-	rb_fdset_t fdreads;
+	fd_set fdreads;
 
-	rb_fd_init(&fdreads);
-	rb_fd_set(epfd, &fdreads);
+	FD_ZERO(&fdreads);
+	FD_SET(epfd, &fdreads);
 
-	if ((ret = rb_thread_fd_select(epfd + 1, &fdreads, NULL, NULL, &tv)) < 1) {
+	if ((ret = rb_thread_select(epfd + 1, &fdreads, NULL, NULL, &tv)) < 1) {
```

Comment:
Thanks for all the comments inline.  This one is especially helpful, because it explains something that looks at first glance like a reversion.

:+1: to all of it.